src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -125,6 +120,17 @@ private[impl] class LaunchQueueActor(
       sender() ! None
   }
 
+  private[this] def receiveInstanceUpdateToSuspendedActor: Receive = {
+    case update: InstanceChange if suspendedLauncherPathIds(update.runSpecId) =>
+      // Do not defer. If an AppTaskLauncherActor restarts, it retrieves a new task list.
+      // If we defer this, there is a potential deadlock (resolved by timeout):
+      //   * AppTaskLauncher waits for in-flight tasks
+      //   * TaskOp gets processed and one of the update steps calls this here
+      //   * ... blocked until timeout ...
+      //   * The task launch notification (that the AppTaskLauncherActor waits for) gets sent to the actor
+      sender() ! None
```

Comment:
Oh- I should have been more explicit- it just wasn't clear in this particular call site what the inner type of the option is.